components/tidb_query_datatype/src/codec/data_type/chunked_vec_enum.rs

```diff
@@ -0,0 +1,142 @@
+use super::bit_vec::BitVec;
+use super::{ChunkRef, ChunkedVec, UnsafeRefInto};
+use super::{Enum, EnumRef};
+use crate::impl_chunked_vec_common;
+use std::sync::Arc;
+use tikv_util::buffer_vec::BufferVec;
+
+#[derive(Debug, Clone)]
+pub struct ChunkedVecEnum {
+    data: Arc<BufferVec>,
```

Comment:
We must have a way to set `data` in `ChunkedVecEnum`. I think we could either add `set_data` function, or use enum data from the first element being pushed.

Reply:
Maybe we can decide what to do here *(imepment it in another PR)* after we have decided how to construct `ChunkedVecEnum` from rpc?

Reply:
Yes, please add a `TODO` in `ChunkedVecEnum` docs.
```
/// `ChunkedVecEnum` stores enum in a compact way.
/// TODO: add way to set enum column data
#[derive(...)]
pub struct ChunkedVecEnum 
```